install.bat

```diff
@@ -0,0 +1,121 @@
+@echo off
+
+set version=0.20.0
+
+if "%PROCESSOR_ARCHITECTURE%"=="x86" (
+    set binary_arch=386
+) else (
+    set binary_arch=amd64
+)
+
+set fzf_base=%cd%
+
+rem Try to download binary executable
+call :download fzf-%version%-windows_%binary_arch%.zip
+echo.
+echo For more information, see: https://github.com/junegunn/fzf
+exit /b 0
+
+rem check binary
+:check_binary
+echo     - Checking fzf executable ...
+for /f "usebackq tokens=1" %%i in (`"%fzf_base%\bin\fzf.exe" --version`) do set output=%%i
+if errorlevel 1 (
+    echo Error: %output%
+    set binary_error="Invalid binary"
+) else (
+    if %version% neq %output% (
+        echo version: %output% != %version%
+        set binary_error="Invalid version"
+    ) else (
+        echo version: %output%
+        set binary_error=""
+        exit /b 0
+    )
+)
+del "%fzf_base%"\bin\fzf.exe
+exit /b 1
+
+rem try to use curl to download the file
+:try_curl
+set temp=%TMP%\fzf.zip
+where curl >nul 2>&1
+if errorlevel 1 (
+    exit /b 1
+) else (
+    where 7z >nul 2>&1
+    if errorlevel 1 (
+        rem can't find 7-zip
+        where winrar >nul 2>&1
+        if errorlevel 1 (
+            rem also can't find WinRAR
+            echo Please intall 7-zip or WinRAR or put them into PATH if they has been installed!
+        ) else (
+            curl -fLo %temp% %~1 && winrar x %temp% && del /s /q %temp%
+        )
+    ) else (
+        curl -fLo %temp% %~1 && 7z -y x %temp% && del /s /q %temp%
+    )
+    exit /b 0
+)
+goto :eof
+
+rem try to use wget to download the file
+:try_wget
+set temp=%TMP%\fzf.zip
+where wget >nul 2>&1
+if errorlevel 1 (
+    exit /b 1
+) else (
+    where 7z >nul 2>&1
+    if errorlevel 1 (
+        rem can't find 7-zip
+        where winrar >nul 2>&1
+        if errorlevel 1 (
+            rem also can't find WinRAR
+            echo Please intall 7-zip or WinRAR or put them into PATH if they has been installed!
+        ) else (
+            wget -O %temp% %~1 && winrar x %temp% && del /s /q %temp%
+        )
+    ) else (
+        wget -O %temp% %~1 && 7z -y x %temp% && del /s /q %temp%
+    )
+    exit /b 0
+)
+goto :eof
```

Comment:
The code for wget and curl look more or less the same. Can we refactor the code and remove the repetition?

Reply:
@junegunn , I have simple the batch file by the commit ba2c4fda9d1d43d854f33ec2c3d98abe5cc77a58. The way to extract file was change to use one powershell command but not try to find out 7-zip or WinRAR on Windows. Please review the change on the function try_curl and try_wget.

Reply:
> Which implement way you prefer, cmd or powershell?

I'm not a Windows user and have no prior experience with either. Which one do you prefer? :)

It looks like the PowerShell version doesn't require wget or curl, which definitely is a plus, right?